packages/website/pages/docs/concepts/architecture-considerations.md

```diff
@@ -42,5 +42,34 @@ Meanwhile, we're continuing to work on a flexible public key based authenticatio
+
+When displaying an NFT that uses IPFS to link to off-chain data, you have several options for retrieval.
+
+### HTTP gateways
+
+IPFS is built around a peer-to-peer content sharing protocol that allows any computer to provide content to anyone that requests it. This provides [many benefits][concepts-decentralized-storage], but not all computing environments can fully support the peer-to-peer networking paradigm.
+
+While some browsers like [Brave](https://brave.com) offer [native IPFS support][brave-ipfs], other browsers are restricted to HTTP and its related protocols, which makes engaging directly with the peer-to-peer protocol more difficult.
+
+As a platform operator, you can support all browsers by using an IPFS gateway to serve NFTs to your users via HTTP. Because NFT.Storage and other services following [best practices for NFT data on IPFS][ipfs-docs-nft-best-practices] use `ipfs://` URLs instead of `https://`, your web application should support re-writing IPFS URLs to target the gateway of your choice.
+
+See the section on [HTTP gateway URLs](https://docs.ipfs.io/how-to/best-practices-for-nft-data/#http-gateway-url) in the [best practices documentation][ipfs-docs-nft-best-practices] to learn about the structure of gateway URLs and how to create them from an `ipfs://` URL.
+
+#### Gateways and centralization
+
+It's important to note that gateways are a [point of centralization](https://docs.ipfs.io/concepts/ipfs-gateway/#centralization) that is under the control of the gateway operator. This is one reason why we strongly recommend storing gateway-agnostic `ipfs://` URLs in NFT metadata and especially in on-chain records, so that links aren't "tied" to a single provider.
+
+As a platform operator, you can direct users to any IPFS gateway, including [public gateways][public-gateway-checker] run by [Protocol Labs](https://protocol.ai) and other organizations in the IPFS ecosystem.
+
+Because public gateways are a shared resource, you may prefer to [run your own dedicated gateway](https://blog.stacktical.com/ipfs/gateway/dapp/2019/09/21/ipfs-server-google-cloud-platform.html) to provide a more consistent user experience. Thanks to the flexibility of content addressing, you can direct your platforms users to your own gateway in your application layer. Anyone not using your platform will still be able to fetch the data from any public gateway, but your users will get improved latency and reliability.
+
+Another potential optimization is to store redundant copies of your NFT data on a traditional storage service and deliver them via HTTP. This side-steps the IPFS retrieval entirely and may be more performant in some cases, but you'll need to do some bookkeeping to keep track of the HTTP locations of each piece of data. When pursuing this option, we strongly recommend exposing the original `ipfs://` URLs to your users, so that they have multiple options for retrieval and aren't dependent on your custom HTTP service.
```

Comment:
I fail to understand how is this different from running your own gateway.

Reply:
lol, good point! @dchoi27 what do you think? I think I added this from your initial outline, but maybe we don't want to recommend people rolling their own gateways.

Reply:
i guess it depends on what is meant by "gateway" - one option refers to running your own IPFS nodes, the other is just using a mapping of CIDs to HTTP locations where you're not broadcasting what CIDs you have, but just know how to resolve HTTP requests for CIDs that you do have (kind of like Opensea does today)

I think it's fine to mention that folks can run their own gateways. We just have to provide a performant enough experience where it's not a good choice.